### PR TITLE
Cisco umbrella playbooks

### PR DESCRIPTION
**Pull request by vu-socprime**
## Proposed Changes

  - add CiscoUmbrellaEnforcementAPI custom logic app connector
  - add CiscoUmbrellaInvestigateAPI custom logic app connector
  - add CiscoUmbrellaManagementAPI custom logic app connector
  - add CiscoUmbrellaNetworkDeviceManagementAPI custom logic app connector
  - add CiscoUmbrella-BlockDomain playbook
  - add CiscoUmbrella-GetDomainInfo playbook
  - add CiscoUmbrella-AssignPolicyToIdentity playbook

### CODE REVIEW

File: Playbooks/CiscoUmbrella/Playbooks/CiscoUmbrella-AssignPolicyToIdentity/readme.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+1. Click the Azure Sentinel connection resource
+2. Click edit API connection
+3. Click Authorize
+4. Sign in
+5. Click Save
+6. Repeat steps for Cisco Umbrella Network Device Management connector API Connection (For authorizing, the key and the secret need to be provided.)
+
+#### b. Configurations in Sentinel
+
+1. In Azure sentinel analytical rules should be configured to trigger an incident. An incident should have *originId* custom entity. OriginId is Umbrella wide unique identifier for this traffic source (origin), it can be obtained from the corresponding field in Cisco Umbrella logs. Check the [documentation](https://docs.microsoft.com/en-us/azure/sentinel/surface-custom-details-in-alerts) to learn more about adding custom entities to incidents.
```
Comment (sarah-yo) [inline Playbooks/CiscoUmbrella/Playbooks/CiscoUmbrella-AssignPolicyToIdentity/readme.md:41]: Remove "en-us" from URL

File: Playbooks/CiscoUmbrella/Playbooks/CiscoUmbrella-BlockDomain/readme.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+1. Click the Azure Sentinel connection resource
+2. Click edit API connection
+3. Click Authorize
+4. Sign in
+5. Click Save
+6. Repeat steps for Cisco Umbrella Enforcement connector API Connection (For authorizing, the key and the secret need to be provided.)
+
+#### b. Configurations in Sentinel
+
+1. In Azure sentinel analytical rules should be configured to trigger an incident with malicious URL. In the *Entity maping* section of the analytics rule creation workflow, malicious URL should be mapped to **Url** identitfier of the **URL** entity type. Check the [documentation](https://docs.microsoft.com/en-us/azure/sentinel/map-data-fields-to-entities) to learn more about mapping entities.
```
Comment (sarah-yo) [inline Playbooks/CiscoUmbrella/Playbooks/CiscoUmbrella-BlockDomain/readme.md:41]: Remove "en-us" from URL

File: Playbooks/CiscoUmbrella/Playbooks/CiscoUmbrella-GetDomainInfo/readme.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+1. Click the Azure Sentinel connection resource
+2. Click edit API connection
+3. Click Authorize
+4. Sign in
+5. Click Save
+6. Repeat steps for Cisco Umbrella Investigate connector API Connection (For authorizing, the API key needs to be provided in the following format: "Bearer YOUR_API_KEY".)
+
+#### b. Configurations in Sentinel
+
+1. In Azure sentinel analytical rules should be configured to trigger an incident. In the *Entity maping* section of the analytics rule creation workflow, malicious URL should be mapped to **Url** identitfier of the **URL** entity type. Check the [documentation](https://docs.microsoft.com/en-us/azure/sentinel/map-data-fields-to-entities) to learn more about mapping entities.
```
Comment (sarah-yo) [inline Playbooks/CiscoUmbrella/Playbooks/CiscoUmbrella-GetDomainInfo/readme.md:40]: Remove "en-us" from URL

**Review by sarah-yo — CHANGES_REQUESTED** [review]
Please remove any "en-us" localisation in URLs.

Please tidy up the grammar, spelling and punctuation in the readmes.

File: Playbooks/CiscoUmbrella/Playbooks/CiscoUmbrella-AssignPolicyToIdentity/readme.md (hunk 1 above)
Comment (vu-socprime) [reply Playbooks/CiscoUmbrella/Playbooks/CiscoUmbrella-AssignPolicyToIdentity/readme.md]: done

File: Playbooks/CiscoUmbrella/Playbooks/CiscoUmbrella-BlockDomain/readme.md (hunk 2 above)
Comment (vu-socprime) [reply Playbooks/CiscoUmbrella/Playbooks/CiscoUmbrella-BlockDomain/readme.md]: done

File: Playbooks/CiscoUmbrella/Playbooks/CiscoUmbrella-GetDomainInfo/readme.md (hunk 3 above)
Comment (vu-socprime) [reply Playbooks/CiscoUmbrella/Playbooks/CiscoUmbrella-GetDomainInfo/readme.md]: done

**Comment by vu-socprime** [comment]
> Please remove any "en-us" localisation in URLs.
> 
> Please tidy up the grammar, spelling and punctuation in the readmes.

@sarah-yo Done

**Comment by v-jayakal** [comment]
@vu-socprime , can you please move these files into solutions folder of Cisco Umbrella? Path for the Cisco Umbrella solution is https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/CiscoUmbrella. Please add a folder called Playbooks and move these files. You can refer other solutions with playbooks (ex: Cisco ISE) for reference.

**Comment by v-jayakal** [comment]
@vu-socprime , can you please move these files into solutions folder of Cisco Umbrella? Path for the Cisco Umbrella solution is https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/CiscoUmbrella. Please add a folder called Playbooks and move these files. You can refer other solutions with playbooks (ex: Cisco ISE) for reference.

**Comment by vu-socprime** [comment]
> Playbooks

@v-jayakal Done

**Comment by vu-socprime** [comment]
> @vu-socprime , can you please move these files into solutions folder of Cisco Umbrella? Path for the Cisco Umbrella solution is https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/CiscoUmbrella. Please add a folder called Playbooks and move these files. You can refer other solutions with playbooks (ex: Cisco ISE) for reference.

@v-jayakal Done

**Comment by lior-tamir** [comment]
Very nice playbooks + Logic Apps connectors!

The action "Get Domain Status and Categorization" is not parsed properly (shows only "body" field")
Same about "Assign a policy to an identity", but in this case depends if it can return success/failure status?
Experience for "block domains" is not ideal, must add a parameter to use, and fields names are not intuitive?

**Review by sreedharande — APPROVED** [review]

**Comment by vu-socprime** [comment]
> The action "Get Domain Status and Categorization" is not parsed properly (shows only "body" field")

@lior-tamir fixed

**Comment by vu-socprime** [comment]
> Same about "Assign a policy to an identity", but in this case depends if it can return success/failure status?

@lior-tamir Added catching of failures

**Comment by vu-socprime** [comment]
> Experience for "block domains" is not ideal, must add a parameter to use, and fields names are not intuitive?

@lior-tamir Could you please specify what parameter should be added? Thanks

**Review by lior-tamir — APPROVED** [review]

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**